ocs_ci/utility/assisted_installer.py

```diff
@@ -0,0 +1,517 @@
+# -*- coding: utf8 -*-
+"""
+This module is for interaction with Assisted Installer API
+"""
+
+import logging
+import os
+import requests
+import time
+import re
+from urllib.parse import urljoin
+
+from ocs_ci.framework import config
+
+from ocs_ci.ocs.exceptions import OpenShiftAPIResponseException
+
+logger = logging.getLogger(__name__)
+
+
+class OpenShiftAPI(object):
+    """
+    Common class for interaction with various OpenShift APIs
+    """
+
+    def __init__(self, environment="production"):
+        """
+        Initialize required variables
+
+        Args:
+            environment (str): Environment defined in configuration (default: production)
+
+        """
+        # TODO: check required configuration
+        self.env = environment
+
+        self._token = None
+        self._token_exp = 0
+
+        # Load configuration
+        # SSO URL for generating API token
+        self.sso_url = config.AUTH["assisted_installer"][self.env]["sso_url"]
+        # Offline token used for generating API token
```

Comment:
for better understanding. Is it a private key, ".pem" file content?

Reply:
No, this is _offline token_ from the OpenShift cluster Manager console used for accessing the API (in fact, it is used for generating another token, which is then used for accessing the API). See [the documentation](https://access.redhat.com/documentation/en-us/assisted_installer_for_openshift_container_platform/2023/html/assisted_installer_for_openshift_container_platform/installing-with-api#installing-the-openshift-cluster-manager-cli_installing-with-api).